src/kbucket.rs

```diff
@@ -1,146 +1,148 @@
-            store: std::collections::HashMap::new(),
-        }
-    }
-    pub fn store(&mut self, key: Identifier, value: StoreValue) {
-        match value {
-            StoreValue::Node(value) => {
-                println!("Store a node");
-                self.add_node(value);
-            }
-            // TODO:
-            StoreValue::Sample(value) => {
-                println!("Store a value");
-                self.add_store();
-            }
-        }
-    }
-    pub fn find_node(&mut self, y: Node) -> Option<Node> {
-        let bucket_index = self.find_bucket(y.node_id);
-        let mut bucket = self.buckets[bucket_index];
-        let result = self.search_bucket(bucket, y);
-
-        match result.0 {
-            true => {
-                println!("Node[bucket_index]: {:?}", bucket[result.1]);
-                let found_node = bucket[result.1];
-                return found_node
-            }
-            false => {
-                println!("Node is not stored");
-                return None
-            }
+            local_node_id,
+            buckets: [Default::default(); MAX_BUCKETS],
+            store: Default::default(),
         }
     }
+    // Protocol's RPCs:
+    // ---------------------------------------------------------------------------------------------------
     // TODO:
-    pub fn find_value() {}
+    // Probes a node to see if it's online
     pub fn ping() {}
-    
-    // Don't expose functions from here down.
-    // ---------------------------------------------------------------------------------------------------
-    
-    //  Add our node to the bucket if it's not already there.
-    pub fn add_node(&mut self, y: Node) {
-        // TODO: Replace these 3 lines w/ find_node().  Kind of complex to do... Maybe later
-        let bucket_index = self.find_bucket(y.node_id);
-        let mut bucket = self.buckets[bucket_index];
-        let result = self.search_bucket(bucket, y);
 
-        match result.0 {
-            // Node was already stored
-            true => {
-                println!("Node was already stored");
-                return 
-            }
-            // Node wasn't already stored
-            false => {
-                bucket[result.1] = Some(y);
-                self.buckets[bucket_index] = bucket;
-                println!("Node is now stored in routing table");
-                return
+    /// "The most important procedure a Kademlia participant must perform is to locate
+    /// the k closest nodes to some given node ID"
+    ///     - Kademlia Paper
+    ///
+    /// Recieves an id request and returns node information on nodes within
+    /// *its closest bucket* (instead of k-closest nodes) to that id.
+    pub fn find_node(&mut self, id: Identifier) -> FindNodeResult {
+        let result = self.search_table(id);
+        let mut bucket = self.buckets[result.bucket_index];
+
+        if result.found {
+            // Returns Node
```

Comment:
generally only want to leave comments if they add something beyond what the code is clearly communicating...